mitmproxy/addons/tlsconfig.py

```diff
@@ -225,6 +254,18 @@ def tls_start_server(self, tls_start: tls.TlsData) -> None:
             alpn_protos=tuple(server.alpn_offers),
         )
 
+        if ctx.options.tls_strategy is TlsStrategy.MIRROR.name:
+            # Set TLS1.3 ciphers
+            # Maybe there is a better way, but I suck at python :)
```

Comment:
Why? (Sorry, but I'm not a python coder, so I'd like to know)
What if pyopenssl merges the pull request where this one is from?

Reply:
I dislike
```python
ssl_ctx.set_ciphersuites = cs.set_ciphersuites
```
because it monkeypatches a new method onto an existing object. That can be very confusing to debug down the line, especially when pyOpenSSL merges a slightly different implementation for `set_ciphersuites`. You would expect `ssl_ctx.set_ciphersuites` to be what pyOpenSSL's provides, but we'd be secretly overwriting it here then.
 
It's more straightforward if we skip the pyOpenSSL wrapper and directly interact with the OpenSSL API:
```python
SSL._openssl_assert(
    SSL._lib.SSL_CTX_set_ciphersuites(ssl_ctx._context, ciphersuites) == 1
)
```